packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts

```diff
@@ -282,19 +282,22 @@ export class ZoweDatasetNode extends ZoweTreeNode implements IZoweDatasetTreeNod
             options.attributes = true;
             let label: string;
             if (contextually.isSessionNotFav(this)) {
-                responses.push(
-                    await ZoweExplorerApiRegister.getMvsApi(cachedProfile).dataSetsMatchingPattern(
-                        // remove duplicate patterns
-                        [
-                            ...new Set(
-                                this.pattern
-                                    .toUpperCase()
-                                    .split(",")
-                                    .map((p) => p.trim())
-                            ),
-                        ]
-                    )
-                );
+                const dsPatterns = [
+                    ...new Set(
+                        this.pattern
+                            .toUpperCase()
+                            .split(",")
+                            .map((p) => p.trim())
+                    ),
+                ];
+                const mvsApi = ZoweExplorerApiRegister.getMvsApi(cachedProfile);
+                if (mvsApi.dataSetsMatchingPattern) {
+                    responses.push(await mvsApi.dataSetsMatchingPattern(dsPatterns));
+                } else {
+                    for (const dsp of dsPatterns) {
+                        responses.push(await mvsApi.dataSet(dsp));
```

Comment:
We might want to add patch coverage for this scenario to verify the correct function is called based on this condition, but your logic here should definitely handle that.

Reply:
Added such test to make sure that the old API is still being called.
Sorry for the delay